feat(app): Add an option to clear labware offsets

SyntaxColoring opened the pull request:
## Overview

This adds an option to the device reset pages, on both the ODD and desktop app, to delete all of the robot's stored labware offsets.

Goes towards EXEC-1281.

## Test Plan and Hands on Testing

* [x] Try it on a Flex with the desktop app and ODD.

## Changelog

Add a "Clear labware offset data" option, with wording and grouping as shown in Figma:

<img width="512" alt="Screenshot 2025-03-28 at 12 26 43 PM" src="https://github.com/user-attachments/assets/1fd5323a-cd7e-449f-9094-992b89f4bef1" />
<img width="512" alt="Screenshot 2025-03-28 at 12 21 46 PM" src="https://github.com/user-attachments/assets/6a326c52-150b-47e3-949c-de7172bc1d46" />

(Note that Figma also shows some other changes that haven't happened yet, like a new "Clear deck configuration" option and splitting "Clear protocol run history" into "Clear protocol run history" and "Clear all data stored on the robot server".)

## Review requests

See GitHub review comments below.
 
## Risk assessment

Medium at least because of the rxjs stuff.

SyntaxColoring committed:
Wire up the new option to actually issue an HTTP request.

SyntaxColoring committed:
Display the new option.

sfoster1 reviewed (approved):
Yeah looks good!

SyntaxColoring committed:
Update unit tests.

SyntaxColoring commented on app/src/organisms/Desktop/Devices/RobotSettings/RobotSettingsAdvanced.tsx:
```diff
@@ -96,7 +96,11 @@ export function RobotSettingsAdvanced({
 
   const [isRobotReachable, setIsRobotReachable] = useState<boolean>(reachable)
-  const [resetOptions, setResetOptions] = useState<ResetConfigRequest>({})
+  // todo(mm, 2025-03-27): This duplicates the default resetOptions inside DeviceResetSlideout.
+  const [resetOptions, setResetOptions] = useState<ResetConfigRequest>({
+    resetLabwareOffsets: false,
+    settingsResets: {},
+  })
```
Any thoughts for fixing this?

It sort of smells to me like we ought to combine this `useState` with `showDeviceResetModal`. So it'd be `{show: false} | {show: true, resetOptions: ResetConfigRequest}`, and the default would be `{show: false}`.

SyntaxColoring commented on app/src/redux/robot-admin/epic/resetConfigEpic.ts:
This file is the hairy and high-risk part. I learned just enough of rxjs in a rush for this, so there's a lot of room for me to have gotten something wrong. For example, returning stale results, or mishandling errors.

SyntaxColoring committed:
Fix syntax error and lint.

SyntaxColoring committed:
More test updates.

shlokamin commented on app/src/redux/robot-admin/epic/resetConfigEpic.ts:
```diff
@@ -39,14 +70,22 @@ const mapResponseToAction: ResponseToActionMapper<ResetConfigAction> = (
+      // Note that HTTP-level request failures are, at the rxjs level, values instead of errors,
+      // so this forkJoin will (correctly) wait for every HTTP request to complete instead of
+      // bailing as soon as any of them fails.
+      forkJoin(
+        mapActionToRequests(action).map(request => fetchRobotApi(host, request))
+      ).pipe(map(responseGroup => mapResponsesToAction(responseGroup, action)))
```
dear god. i really think we should consider ripping out this epic all together and setting up api client + react api client functions + hooks to reset settings (see `updateRobotSetting` and `useUpdateRobotSettingMutation`)

this just seems so unnecessarily complicated, and at first glance i don't think a refactor would be too bad.

shlokamin replied:
but also bless you @SyntaxColoring for getting into this

SyntaxColoring committed:
Fix resetConfigEpic unit tests.

The 'r' -> '(r|)' thing is funny. All of our tests, this whole time, were using a mock of fetchRobotApi that never signaled completion after emitting its value. My code is apparently the first time this has actually mattered, because I used forkJoin, and forkJoin only returns after all of its arguments complete. So, update the mock to signal completion, like the real fetchRobotApi does.
